pkg/rtc/mediatracksubscriptions.go

```diff
@@ -511,8 +511,11 @@ func (t *MediaTrackSubscriptions) UpdateQualityChange(force bool) {
 		return
 	}
 
-	if (t.maxSubscribedQuality != livekit.VideoQuality_OFF && maxSubscribedQuality != livekit.VideoQuality_OFF) &&
-		t.maxSubscribedQuality > maxSubscribedQuality && !force {
+	// if quality comes down(or become OFF), delay notify to publisher
+	if (t.maxSubscribedQuality != livekit.VideoQuality_OFF) &&
+		(t.maxSubscribedQuality > maxSubscribedQuality || maxSubscribedQuality == livekit.VideoQuality_OFF) &&
+		t.params.VideoConfig.DynacastPauseDelay > 0 && !force {
+
 		t.params.Logger.Debugw("throttle quality change", "from", t.maxSubscribedQuality, "to", maxSubscribedQuality)
 		t.maxQualityLock.Unlock()
 		t.maxSubscribedQualityDebounce(func() {
```

Comment:
Maybe not `force` here. Let it recalculate after a delay.

Reply:
without force, it may go to a loop that we can't update t.masSubscribedQuality